app/models/user.rb

```diff
@@ -0,0 +1,56 @@
+# == Schema Information
+#
+# Table name: users
+#
+#  id                     :bigint           not null, primary key
+#  confirmation_sent_at   :datetime
+#  confirmation_token     :string
+#  confirmed_at           :datetime
+#  current_sign_in_at     :datetime
+#  current_sign_in_ip     :inet
+#  email                  :string           default(""), not null
+#  encrypted_password     :string           default(""), not null
+#  full_name              :string           not null
+#  gender                 :integer          default("other"), not null
+#  last_sign_in_at        :datetime
+#  last_sign_in_ip        :inet
+#  provider               :string           default("email"), not null
+#  remember_created_at    :datetime
+#  reset_password_sent_at :datetime
+#  reset_password_token   :string
+#  sign_in_count          :integer          default(0), not null
+#  tokens                 :json
+#  uid                    :string           default(""), not null
+#  unconfirmed_email      :string
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#
+# Indexes
+#
+#  index_users_on_email                 (email) UNIQUE
+#  index_users_on_reset_password_token  (reset_password_token) UNIQUE
+#  index_users_on_uid_and_provider      (uid,provider) UNIQUE
+#
+class User < ApplicationRecord
+  include DeviseTokenAuth::Concerns::User
+
+  devise :database_authenticatable, :registerable, :recoverable, :rememberable,
+         :validatable, :confirmable, :trackable
+
+  enum gender: { male: 0, female: 1, other: 2 }
+
+  before_validation :init_uid
+
+  validates :email, presence: true, uniqueness: true
```

Comment:
when you're using the same validation for differents attributes you can write it on the same line:
`validates :full_name, :gender, presence: true`